engine/execution/ingestion/engine.go

```diff
@@ -558,7 +561,11 @@ func (e *Engine) executeBlock(ctx context.Context, executableBlock *entity.Execu
 
 	isExecutedBlockSealed := executableBlock.Block.Header.Height <= lastSealed.Height
 	broadcasted := false
-	if !isExecutedBlockSealed {
+	stakedAtBlock, err := e.checkStakedAtBlock(executableBlock.ID())
```

Comment:
we could move this inside of the `if !isExecutedBlockSealed` so that, if the block has already sealed, we don't need to check whether I'm staked

Reply:
Another thing is we could cache the block id [here](https://github.com/onflow/flow-go/pull/400/files#diff-e5c0dea70cf688717060471583da13cc5269a6f661e955613ed7ef698354cc1aR522),
so that we can read the cached block id here, [here](https://github.com/onflow/flow-go/pull/400/files#diff-e5c0dea70cf688717060471583da13cc5269a6f661e955613ed7ef698354cc1aR535) and [here](https://github.com/onflow/flow-go/pull/400/files#diff-e5c0dea70cf688717060471583da13cc5269a6f661e955613ed7ef698354cc1aR540)

Reply:
`ExecutableBlock` caches `blockID` internally